pkg/operator/ceph/cluster/watcher.go

```diff
@@ -53,61 +56,61 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 	if !ok {
 		return false
 	}
+	// Check if the node have OSDs already present on it, reconcile if OSDs are not present
+	if !nodesCheckedForReconcile.Contains(node.Name) {
```

Comment:
better to immediately return, then we can avoid the deeper code blocks
```suggestion
	if nodesCheckedForReconcile.Contains(node.Name) {
		return false
	}
```

Reply:
yes right, but if don't check for positive case separately(`if nodesCheckedForReconcile.Contains(node.Name) {`) then also execution will directly jump to statement `return false` on line 121, so it is needed?

Reply:
Here on line 60 (and similarly on line 96 below) deeper code blocks are created. I'm suggesting that by returning immediately with the opposite condition, no deeper code blocks would be needed. If we can avoid the deeper code blocks, the code is more easily maintainable IMO.

Reply:
looks like you missed to remove `!` in the if statement. We need to return false if node is already present in the `nodesCheckedForReconcile`.

```suggestion
	if nodesCheckedForReconcile.Contains(node.Name) {
```